### PR TITLE
[fix]: [CDS-72568]: Handling service input set as runtime execution input

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Earlier we were only supporting the execution inputs values as leaf fields. So, now we modified the code to handle those cases when execution input field can itself be a complete json object. 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50396)

### CODE REVIEW

**Comment by lovish1999** [comment]
trigger messagecheck

**Comment by lovish1999** [comment]
trigger uts

**Comment by brijeshdhakar** [comment]
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java` line 46 at r1](https://reviewable.io/reviews/harness/harness-core/50396#-N_DqdhbAfspWLpSFBds:-N_DqdhbAfspWLpSFBdt:b-dvssd2) ([raw file](https://github.com/harness/harness-core/blob/c5a69e4bc7db085ad19f05000446e15b7157a2b1/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java#L46)):*
> ```Java
>         if (rawInputValue == null) {
>           YamlNode userInputYamlNode = new YamlNode(userInput);
>           YamlNode executionInputYamlNode = YamlNodeUtils.goToPathUsingFqn(userInputYamlNode, key.getExpressionFqn());
> ```

Can you add comments why and when the rawInputValue is null while reading from FQNMap and its not null while reading from from jsonNode `YamlNodeUtils.goToPathUsingFqn(userInputYamlNode, key.getExpressionFqn());`?

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50396)* status: 3 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java` line 46 at r1](https://reviewable.io/reviews/harness/harness-core/50396#-N_DqdhbAfspWLpSFBds:-N_Y99ma2OosIQw65oDR:bbfgaho) ([raw file](https://github.com/harness/harness-core/blob/c5a69e4bc7db085ad19f05000446e15b7157a2b1/pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java#L46)):*
<details><summary><i>Previously, brijeshdhakar (Brijesh Dhakar) wrote…</i></summary><blockquote>

Can you add comments why and when the rawInputValue is null while reading from FQNMap and its not null while reading from from jsonNode `YamlNodeUtils.goToPathUsingFqn(userInputYamlNode, key.getExpressionFqn());`?
</blockquote></details>

I modified the logic to use the json map conversion in the main execution helper class itself. So this comment is not required now.

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java [hunk 1]
```diff
@@ -38,8 +45,19 @@ public Map<String, Object> getExecutionInputMap(JsonNode inputTemplate, JsonNode
     templateFqnToValueMap.keySet().forEach(key -> {
       String value = templateFqnToValueMap.get(key).toString().replace("\\\"", "").replace("\"", "");
       if (NGExpressionUtils.matchesExecutionInputPattern(value)) {
-        Object rawInputValue = fullUserInputMap.get(key);
-        Object inputValue = JsonNodeUtils.getValueFromJsonNode(rawInputValue);
+        Object inputValue = new Object();
+        String keyFieldName = key.getFieldName();
+        if (isNotEmpty(keyFieldName) && ENTITY_INPUTS_FIELD_NAMES_SET.contains(keyFieldName)) {
```
Comment (tchaurasiya) [inline pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java:50]: lets add comment, for utelisation of this block to handle execution input at non-leaf nodes.

**Comment by lovish1999** [comment]
trigger uts

File: pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java (hunk 1 above)
Comment (lovish1999) [reply pipeline-service/modules/orchestration/src/main/java/io/harness/engine/execution/ExecutionInputServiceHelper.java:56]: done

**Review by shalini-agr — APPROVED** [review]

**Review by tchaurasiya — APPROVED** [review]

**Comment by lovish1999** [comment]
trigger review

**Pull request merged**